### ISSUE TITLE
TITANIC PREDICTION CODE(KNN 분류 알고리즘 사용)

### ISSUE THREAD

**Issue by yangwonsuck** [yangwonsuck/python_code#2]
# 기본 라이브러리 불러오기
import pandas as pd
import seaborn as sns

#load_dataset 함수를 사용하여 데이터프레임으로 변환
df=sns.load_dataset('titanic')
print(df.head())
print('\n')

#IPython 디스플레이 설정 - 출력할 열의 개수 한도 늘리기
pd.set_option('display.max_columns',15)
print(df.head())

#e데이터 자료형 확인
print(df.info())

#NaN 값이 많은 deck 열 삭제, embarked와 내용이 겹치는 embark_town 열 삭제
rdf=df.drop(['deck','embark_town'],axis=1)
print(rdf.columns.values)

# age 열에 나이 데이터가 없는 모든 행 삭제 - age 열 (891개 중 177개의 NaN 값)
rdf=rdf.dropna(subset=['age'],how='any',axis=0)
print(len(rdf))

# embarked 열의 NaN 값을 승선도시 중에서 가장 많이 출현한 값으로 치환하기
most_freq=rdf['embarked'].value_counts(dropna=True).idxmax()
print(most_freq)
print('\n')

print(rdf.describe(include='all'))
print('\n')

rdf['embarked'].fillna(most_freq,inplace=True)

#분석에 활용할 열(속성) 선택
ndf=rdf[['survived','pclass','sex','age','sibsp','parch','embarked']]
print(ndf.head())

#원핫인코딩-범주형 데이터를 모형이 인식할 수 있도록 숫자형으로 변환
onehot_sex=pd.get_dummies(ndf['sex'])
ndf=pd.concat([ndf,onehot_sex],axis=1)

onehot_embarked=pd.get_dummies(ndf['embarked'],prefix='town')
ndf=pd.concat([ndf,onehot_embarked],axis=1)

ndf.drop(['sex','embarked'],axis=1,inplace=True)
print(ndf.head())

#속성(변수) 선택
X=ndf[['pclass','age','sibsp','parch','female','male','town_C','town_Q','town_S']] #설명 변수 X
y=ndf['survived'] #예측 변수 Y

# 설명 변수 데이터를 정규화
from sklearn import preprocessing
X=preprocessing.StandardScaler().fit(X).transform(X)

# train data와 test data로 구분(7:3 비율)
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=10)

print('훈련 데이터: ',X_train.shape)
print('검증 데이터: ',X_test.shape)

# sklearn 라이브러리에서 KNN 분류 모형 가져오기
from sklearn.neighbors import KNeighborsClassifier

# 모형 객체 생성(k=5로 설정)
knn=KNeighborsClassifier(n_neighbors=5)

# train data를 가지고 모형 학습
knn.fit(X_train,y_train)

#test data를 가지고 y_hat을 예측(분류)
y_hat=knn.predict(X_test)

print(y_hat[0:10])
print(y_test.values[0:10])

from sklearn import metrics
knn_matrix=metrics.confusion_matrix(y_test,y_hat)
print(knn_matrix)

#모형 성능 평가 - Confsion Matrix 계산
knn_report=metrics.classification_report(y_test,y_hat)
print(knn_report)
